### PR TITLE
First draft to support unexpected enum values

### PR DESCRIPTION
**Pull request by brice-laurencin**
resolves quarkiverse/quarkus-openapi-generator#692

This is a first draft, I would follow the guidelines if you thinks this feature is useful.
I still have an issue with numbered enum that I need to dig into, but I'd like to have a first feedback on this version.

Please make sure that your PR meets the following requirements:

- [x] You have read the [contributors guide](CONTRIBUTING.md)
- [x] Your code is properly formatted according to [our code style](CONTRIBUTING.md#code-style)
- [x] Pull Request title contains the target branch if not targeting main: `[0.9.x] Subject`
- [x] Pull Request contains link to the issue
- [x] Pull Request contains link to any dependent or related Pull Request
- [x] Pull Request contains description of the issue
- [x] Pull Request does not include fixes for issues other than the main ticket

### CODE REVIEW

**Comment by mcruzdev** [comment]
Hi, @brice-laurencin Thank you for your pull request, I think that it is an in-progress pull request, right?

Would be great to have a documentation for our users, to know how to use the new property, WTY?

**Review by mcruzdev — COMMENTED** [review]
Hi, @brice-laurencin Thank you for your pull request, I think that it is an in-progress pull request, right?

Would be great to have a documentation for our users, to know how to use the new property, WTY?

**Comment by brice-laurencin** [comment]
Definitely.
This is a first draft to know if I should push the effort further, if you (the maintainers of the extension) think this is valuable.

**Review by ricardozanini — APPROVED** [review]

File: client/deployment/src/main/java/io/quarkiverse/openapi/generator/deployment/CodegenConfig.java [hunk 1]
```diff
@@ -20,6 +20,10 @@ public class CodegenConfig extends GlobalCodegenConfig {
 
     public static final String API_PKG_SUFFIX = ".api";
     public static final String MODEL_PKG_SUFFIX = ".model";
+
+    public static final String ADDITIONAL_ENUM_TYPE_UNEXPECTED_MEMBER_NAME_DEFAULT = "UNEXPECTED";
+    public static final int ADDITIONAL_ENUM_TYPE_UNEXPECTED_MEMBER_INT_VALUE_DEFAULT = -1;
```
Comment (hbelmiro) [inline client/deployment/src/main/java/io/quarkiverse/openapi/generator/deployment/CodegenConfig.java:25]: This is never used and can be removed.

File: client/deployment/src/main/java/io/quarkiverse/openapi/generator/deployment/codegen/OpenApiGeneratorCodeGenBase.java [hunk 2]
```diff
@@ -1,8 +1,6 @@
 package io.quarkiverse.openapi.generator.deployment.codegen;
 
-import static io.quarkiverse.openapi.generator.deployment.CodegenConfig.getGlobalConfigName;
-import static io.quarkiverse.openapi.generator.deployment.CodegenConfig.getSanitizedFileName;
-import static io.quarkiverse.openapi.generator.deployment.CodegenConfig.getSpecConfigName;
+import static io.quarkiverse.openapi.generator.deployment.CodegenConfig.*;
```
Comment (hbelmiro) [inline client/deployment/src/main/java/io/quarkiverse/openapi/generator/deployment/codegen/OpenApiGeneratorCodeGenBase.java:3]: Please don't use star imports.

File: client/integration-tests/enum-unexpected/src/test/java/io/quarkiverse/openapi/generator/it/EnumUnexpectedTest.java [hunk 3]
```diff
@@ -0,0 +1,113 @@
...
+            throws JsonProcessingException {
+        org.openapi.quarkus.with_enum_unexpected_yaml.model.Echo actualEcho = objectMapper.readValue("{ \"msgType\": \"NOPE\"}",
+                org.openapi.quarkus.with_enum_unexpected_yaml.model.Echo.class);
+        MatcherAssert.assertThat(actualEcho.getMsgType(),
+                is(org.openapi.quarkus.with_enum_unexpected_yaml.model.Echo.MsgTypeEnum.UNEXPECTED));
+    }
+
+    @Test
+    void when_additional_enum_type_unexpected_member_is_false_should_fail_parsing_unknown_values()
+            throws JsonProcessingException {
```
Comment (hbelmiro) [inline client/integration-tests/enum-unexpected/src/test/java/io/quarkiverse/openapi/generator/it/EnumUnexpectedTest.java:74]: Exception 'com.fasterxml.jackson.core.JsonProcessingException' is never thrown in the method 

```suggestion
    void when_additional_enum_type_unexpected_member_is_false_should_fail_parsing_unknown_values() {
```

File: docs/modules/ROOT/pages/client.adoc [hunk 4]
```diff
@@ -50,6 +50,11 @@ include::./includes/circuit-breaker.adoc[leveloffset=+1, opts=optional]
 
 include::./includes/sending-multipart-formdata.adoc[leveloffset=+1, opts=optional]
 
+[[Supporting unexpected enums values]]
```
Comment (hbelmiro) [inline docs/modules/ROOT/pages/client.adoc:53]: Block ID pattern invalid, must not contain spaces or tabs

**Review by hbelmiro — COMMENTED** [review]
Good job @brice-laurencin. Thank you.
I left a few minor comments.

**Comment by brice-laurencin** [comment]
> Good job @brice-laurencin. Thank you. I left a few minor comments.

Thanks a lot @hbelmiro, I'll have those fixed this week.

**Review by hbelmiro — APPROVED** [review]

**Comment by hbelmiro** [comment]
@all-contributors add @brice-laurencin for code.

**Comment by brice-laurencin** [comment]
Thanks a lot @hbelmiro! :blush:

**Comment by brice-laurencin** [comment]
Hello @hbelmiro, it seems this code is gone from main with the new 3.0.0-SNAPSHOT version.

any reason for this?

(I just saw that as I'm trying to contribute a PR for https://github.com/quarkiverse/quarkus-openapi-generator/issues/520 as I would very much need this)

**Comment by ricardozanini** [comment]
@brice-laurencin have you looked at `client` module? We've moved things around to accommodate the new `server` module.

**Comment by brice-laurencin** [comment]
I have, and it seems gone from the templates and the integration tests.
Searching for the config entries I added for this PR gives me nothing.

**Comment by hbelmiro** [comment]
I don't get it @brice-laurencin.
I can see the template here: https://github.com/quarkiverse/quarkus-openapi-generator/blob/1a5ab2d132c2d5f01ca1986a1c73b2628434fb47/client/deployment/src/main/resources/templates/libraries/microprofile/additionalEnumTypeUnexpectedMember.qute
And the tests here: https://github.com/quarkiverse/quarkus-openapi-generator/blob/1a5ab2d132c2d5f01ca1986a1c73b2628434fb47/client/integration-tests/enum-unexpected/src/test/java/io/quarkiverse/openapi/generator/it/EnumUnexpectedTest.java
What files are you missing exactly?

**Comment by brice-laurencin** [comment]
Oh sorry. PEBKAC all the way here, I messed up between `origin` and `upstream` :disappointed:

**Pull request merged**